### DISCUSSION TITLE
HA API keeps re-connecting

### DISCUSSION THREAD

**Discussion by flexmatics** [nliaudat/floor-heating-controller#1] [Q&A] [answered]
Hi,

Can someone point me to the right direction. 

In board.yaml, default settings present:
![image](https://user-images.githubusercontent.com/79056776/192143959-a1ee0274-ec13-4808-b80e-580110ad9a28.png)

In ESP logs seems that HA keeps re-connecting:
![image](https://user-images.githubusercontent.com/79056776/192143982-b3338f14-c52b-4fd2-9cf8-02ce84739980.png)

This also causes climate entity to become unavailable in HA:
![image](https://user-images.githubusercontent.com/79056776/192144024-f39c33ee-8603-46cd-9c54-b3e911280ee7.png)

What could be the reason for this? what could I try to fix this? thanks.

**Comment by nliaudat** [comment]
Could you check your wifi signal ?
Can be got in hassio or by enabling webserver in config.yaml

```
 web_server:
   port: 80  
```

**Comment by flexmatics** [comment]
Seems to be that the signal is ok:
![image](https://user-images.githubusercontent.com/79056776/192155082-470030e3-99b1-49b7-8d7c-ba4ccf6d4482.png)

Other WIFI settings:
![image](https://user-images.githubusercontent.com/79056776/192155141-696a1cd5-075d-462c-b693-8fb93e2b46aa.png)

**Comment by nliaudat** [comment]
Try disabling fast_connect and set [power_save_mode: none](https://esphome.io/components/wifi.html?highlight=power_save_mode#wifi-power-save-mode)

**Reply by nliaudat** [reply]
look at https://github.com/esphome/issues/issues/1237

**Comment by nliaudat** [comment]
2nd idea : disable static IP and look if it's better with dhcp assigned IP. Perhaps you have something misconfigured in your static IP setting

**Comment by flexmatics** [comment]
thanks; will try those ideas step by step.

**Comment by flexmatics** [comment]
Unfortunately still have the same issue:
![image](https://user-images.githubusercontent.com/79056776/192157052-a98528f0-e2ef-4854-a696-e74d3534af06.png)

Latest settings used:
![image](https://user-images.githubusercontent.com/79056776/192157060-0b690514-8f99-4575-97f6-5a03f37bca03.png)

Perhaps ESP board is faulty..weird

**Comment by flexmatics** [comment, marked as answer]
Ok, so it seems I found the issue. Half an hour passed with no disconnects.

The issue was with sub-nets. I have main sub-net (where home assistant is located), 192.168.0.1. And also made additional sub-net for IoT stuff (192.168.2.1).

Moved ESP back to main sub-net (192.168.0.1), the issue seems to be gone. Why is this an issue I don't know nor will I try to find out - as long as it works now..
